devfile/devworkspace-operator
# Support specifying annotations for devworkspace pods in the DWOC

### What does this PR do?
Provides `config.workspace.podAnnotations` field for DWOC to specify annotations for the devworkspace pods

### What issues does this PR fix or reference?
https://github.com/devfile/devworkspace-operator/issues/1270

### Is it tested? How?

1. Deploy DWO by using this catalog source and installing DWO from OperatorHub:
```
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: devworkspace-operator-catalog
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: quay.io/dkwon17/devworkspace-operator-index:annotations-dwoc
  publisher: Red Hat
  displayName: DevWorkspace Operator Catalog
  updateStrategy:
    registryPoll:
      interval: 5m
```
2. Update the `devworkspace-operator-config` DWOC by adding an annotation:
```
config:
  workspace:
    podAnnotations:
      test-annotation: test
```

3. Create a workspace:
```
oc apply -f ./samples/plain.yaml -n openshift-operators 
```
4. Once the workspace pod is created, check the pod annotations and verify that `test-annoation: test` exists:
```
metadata:
  annotations:
    test-annotation: test
    ...
```
### PR Checklist
